### PR TITLE
calibre: cssutils -> css-parser

### PR DESCRIPTION
**Pull request by jethrokuan**
###### Motivation for this change

Updates an altered calibre dependency. Calibre now fails with:

```
ImportError: No module named css_parser.css
```

because of a recent change https://github.com/kovidgoyal/calibre/commit/dd7d8ea3c4f47e315623d882d6b68339da959ab9

This also introduces a new package css-parser, which calibre uses.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Review by dotlambda — CHANGES_REQUESTED** [review]
Adding the python package should be a separate commit.

File: pkgs/development/python-modules/css-parser/default.nix [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+buildPythonPackage rec {
+  pname = "css-parser";
+  version = "1.0.4";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "c7ab355512ae51334ba6791a7e4d553f87bef17ba2026f1cc9bf3b17a7779d44";
+  };
+
+  buildInputs = [ mock ];
```
[the remark this answers is not in the document]
Comment (jethrokuan) [reply pkgs/development/python-modules/css-parser/default.nix]: These were all inherited from the previous package `cssutils`. `css-parser` is just an updated fork of `cssutils` with minor changes, so I assumed nothing much had changed.

**Comment by jethrokuan** [comment]
Taking your reviews into account, but closing this PR to split it into 2.

**Comment by dotlambda** [comment]
No need for two PRs, just two commits.

**Comment by jethrokuan** [comment]
done, thanks for reviewing again @dotlambda

File: pkgs/development/python-modules/css-parser/default.nix (hunk 1 above)
[the remark this answers is not in the document]
Comment (dotlambda) [reply pkgs/development/python-modules/css-parser/default.nix]: I.e. you're not sure whether it belongs there. Please remove it and remember to enable tests (and possibly add `checkInputs`) once a new version is released.

[the remark this answers is not in the document]
Comment (jethrokuan) [reply pkgs/development/python-modules/css-parser/default.nix]: Looks like any dependency on it has been removed. Sorry for that, should've done the due diligence.

File: pkgs/development/python-modules/css-parser/default.nix [hunk 2]
```diff
@@ -0,0 +1,22 @@
+{ lib, buildPythonPackage, fetchPypi, mock }:
```
Comment (dotlambda) [inline pkgs/development/python-modules/css-parser/default.nix]:
```suggestion
{ lib, buildPythonPackage, fetchPypi }:
```

**Comment by dotlambda** [comment]
@GrahamcOfBorg build python2.pkgs.css-parser python3.pkgs.css-parser

**Comment by dotlambda** [comment]
@GrahamcOfBorg build calibre

**Pull request merged**